src/expression.cpp

```diff
@@ -248,6 +248,68 @@ double eval_dbl(const expression &e, const std::unordered_map<std::string, doubl
     return std::visit([&map](const auto &arg) { return eval_dbl(arg, map); }, e.value());
 }
 
+void eval_batch_dbl(const expression &e, const std::unordered_map<std::string, std::vector<double>> &map,
```

Comment:
The convention used elsewhere is that return value parameters are passed as first arguments. Would it be possible to change the prototypes of these and similar functions to take ``retval`` as first parameter?

To be clear, this "rule" applies only to those functions which "formally" would produce a return value but instead take the return value by reference for performance reasons.

Reply:
k